internal/sync/sync.go

```diff
@@ -0,0 +1,332 @@
+		fmt.Sprintf("package %s", path.Base(localDir)),
+		localDir,
+		"*.go")
+
+	if err != nil {
+		glog.V(100).Infof("Failed to refactor file. Exist with error 1")
+		os.Exit(1)
+	}
+
+	for _, importMap := range repo.ReplaceImports {
+		err = refactor(fmt.Sprintf("%q", importMap["old"]), importMap["new"], localDir, "*.go")
+
+		if err != nil {
+			glog.V(100).Infof("Failed to refactor file. Exist with error 1")
+			os.Exit(1)
+		}
+	}
+}
+
+func gitReset(packageName string) {
+	for _, cmdToRun := range [][]string{
+		{"reset", "--", fmt.Sprintf("./pkg/%s", packageName)},
+		{"checkout", "--", fmt.Sprintf("./pkg/%s", packageName)},
+		{"clean", "-d", "-f", fmt.Sprintf("./pkg/%s", packageName)},
+	} {
+		err := execCmd("", "git", cmdToRun)
+
+		if err != nil {
+			glog.Infof("Failed to reset project to it's original state. Exist with error 1")
+			os.Exit(1)
+		}
+	}
+}
+
+func gitClone(localPath string, repo *repo) {
+	glog.V(100).Infof("Cloning repo %s from %s", repo.Name, repo.RepoLink)
+	localDirectory := path.Join(localPath, repo.Name)
+
+	if _, err := os.Stat(localDirectory); !os.IsNotExist(err) {
+		glog.V(100).Infof(
+			"Local directory already exist for the project: %s. Removing directory", repo.Name)
+
+		err := os.RemoveAll(localDirectory)
+
+		if err != nil {
+			glog.V(100).Infof("Failed to remove repo directory %s due to %w exit with error code 1",
+				localDirectory, err)
+			os.Exit(1)
+		}
+	}
+
+	for idx, gitArgs := range [][]string{
+		{"clone", "-n", "--depth=1", "--filter=tree:0", "-b", repo.Branch, repo.RepoLink},
+		{"sparse-checkout", "set", "--no-cone", repo.RemoteAPIDirectory}, {"checkout"}} {
+		directory := localDirectory
+
+		// first cmd is git clone, it is necessary to run this cmd in parents directory in order to create repo's dir.
+		if idx == 0 {
+			directory = localPath
+		}
+
+		err := execCmd(directory, "git", gitArgs)
+
+		if err != nil {
+			glog.V(100).Infof("Failed to clone repo due to cmd error. Exit with error code 1")
+			os.Exit(1)
+		}
+	}
+}
+
+func execCmd(dirName, binary string, args []string) error {
+	glog.V(100).Infof("Executing cmd: %s, with args: %v, in directory: %s", binary, args, dirName)
+
+	cmd := exec.Command(binary, args...)
+
+	if dirName != "" {
+		cmd.Dir = dirName
+	}
+
+	out, err := cmd.Output()
+
+	if err != nil {
+		glog.V(100).Infof("Failed to execute cmd due to %w. Output: %s", err, string(out))
+
+		return err
+	}
+
+	return nil
+}
+
+// NewConfig returns instance of GeneralConfig config type.
+func NewConfig() []repo {
+	var config []repo
+
+	glog.V(100).Infof("Read file")
+
+	err := readFile(&config, "internal/sync/sync-config.yaml")
+
+	if err != nil {
+		glog.V(100).Infof("Error to read file: %w", err)
+
+		return nil
+	}
+
+	return config
+}
+
+func readFile(cfg *[]repo, cfgFile string) error {
+	openedCfgFile, err := os.Open(cfgFile)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		_ = openedCfgFile.Close()
+	}()
```

Comment:
Isn't it the same as this?
```suggestion
	defer openedCfgFile.Close()
```